lib/internal/http2/core.js

```diff
@@ -2702,7 +2699,6 @@ class Http2Server extends NETServer {
   constructor(options, requestListener) {
     super(connectionListener);
     this[kOptions] = initializeOptions(options);
-    this.timeout = kDefaultSocketTimeout;
```

Comment:
```suggestion
    this.timeout = 0;
```

...or if `kDefaultSocketTimeout` is set to `0`, leave the line as-is?